src/stitching/makeRemoteExecutableSchema.ts

```diff
@@ -88,18 +116,18 @@ export default function makeRemoteExecutableSchema({
   });
 }
 
-function createResolver(fetcher: Fetcher): GraphQLFieldResolver<any, any> {
+function createResolver(link: ApolloLink): GraphQLFieldResolver<any, any> {
   return async (root, args, context, info) => {
     const operation = print(info.operation);
     const fragments = Object.keys(info.fragments)
       .map(fragment => print(info.fragments[fragment]))
       .join('\n');
     const query = `${operation}\n${fragments}`;
-    const result = await fetcher({
-      query,
+    const result = await makePromise(execute(link, {
+      query: typeof query === 'string' ? parse(query) : query,
```

Comment:
What does link accept exactly? Cause we have parsed document already above and we print it. If it accepts parsed document, then we should pass it the parsed document.